### PR TITLE
MINOR: KafkaBroker.brokerState should be volatile instead of AtomicReference

### PR DESCRIPTION
**Pull request by ijuma**
We don't need or use the additional functionality provided by
AtomicReference.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
cc @rondagostino

File: core/src/main/scala/kafka/server/KafkaBroker.scala [hunk 1]
```diff
@@ -71,8 +70,11 @@ object KafkaBroker {
 }
 
 trait KafkaBroker extends KafkaMetricsGroup {
+  @volatile private var _brokerState: BrokerState = BrokerState.NOT_RUNNING
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaBroker.scala:73]: Why not using ```@volatile var brokerState: BrokerState = BrokerState.NOT_RUNNING```?

Comment (ijuma) [reply core/src/main/scala/kafka/server/KafkaBroker.scala:73]: Because that would make it mutable for everyone. The current implementation is mutable only within subclasses.

**Review by mumrah — APPROVED** [review]
LGTM

File: core/src/main/scala/kafka/server/KafkaBroker.scala [hunk 2]
```diff
@@ -90,7 +92,7 @@ trait KafkaBroker extends KafkaMetricsGroup {
     explicitMetricName(KafkaBroker.metricsPrefix, KafkaBroker.metricsTypeName, name, metricTags)
   }
 
-  newGauge("BrokerState", () => brokerState.get.value())
+  newGauge("BrokerState", () => brokerState.value)
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaBroker.scala:95]: (this is not related to this PR) it is a bit weird to me that we expose "byte" as "broker state".

**Review by chia7712 — APPROVED** [review]

File: core/src/main/scala/kafka/server/KafkaBroker.scala (hunk 2 above)
Comment (ijuma) [reply core/src/main/scala/kafka/server/KafkaBroker.scala:95]: Right. To change this, it would require a KIP.
